complex/src/lib.rs

```diff
@@ -740,6 +742,107 @@ impl<T> fmt::Binary for Complex<T> where
     }
 }
 
+impl<T> FromStr for Complex<T> where
+    T: FromStr + Num + PartialOrd + Clone + std::fmt::Debug, T::Err: Error + PartialEq
+{
+    type Err = ParseComplexError<T>;
+
+    /// Parses `a +/- bi`; `ai +/- b`; `a`; or `bi` where `a` and `b` are of type `T`
+    fn from_str(s: &str) -> Result<Complex<T>, ParseComplexError<T>>
+    {
+        let imag = match s.rfind('j') {
+            None => 'i',
+            _ => 'j'
+        };
+
+        let mut b = String::with_capacity(s.len());
+        let mut first = true;
+
+        let mut char_indices = s.char_indices();
+        let mut pc = ' ';
+        let mut split_index = s.len();
+
+        loop {
+            match char_indices.next() {
+                Some(t) => {
+                    let i = t.0;
+                    let cc = t.1;
+
+                    if cc == '+' && pc != 'e' && pc != 'E' {
+                        // ignore '+' if part of an exponent
```

Comment:
Rust parses `+` at the beginning of numbers too, so we should allow and include that.  But I'd say still skip the middle `+`, so we don't have to *force* the explicit sign.